src/kernelfunctions/mercer/exponential.jl

```diff
@@ -28,19 +28,20 @@ julia> ExponentialKernel(2.0f0)
 ExponentialKernel{Float32}(2.0)
 ```
 """
-struct ExponentialKernel{T<:AbstractFloat} <: AbstractExponentialKernel{T}
-    α::T
-    function ExponentialKernel{T}(α::Real=T(1)) where {T<:AbstractFloat}
-        @check_args(ExponentialKernel, α, α > zero(T), "α > 0")
-        return new{T}(α)
+struct ExponentialKernel{T<:Real,A} <: AbstractExponentialKernel{T}
+    α::A
```

Comment:
The `A` parameter has a nice property to become either a scalar or vector, and the second big advantage is that it will allow compatibility with `CuArrays` and other more complex arrays if other users want to use the package. But indeed one could restrict `A` to `A<:Union{AbstractVector{T},T}`